### ISSUE TITLE
Card is null in customer source

### ISSUE THREAD

**Issue by BendicoE** [stripe/stripe-dotnet#1090]
I'm having trouble with accessing the card information in a customer source. (I'm using the new sources workflow in Stripe.net.) I create a source with javascript in the frontend and on the server I use StripeSourceService.Get() to access it. The .Card member of the StripeSource object now seems to contain the Card information, with Fingerprint, etc. I then attach this source to a StripeCustomer.

If I now loop through the StripeCustomer.Sources, I can see that the source with that Id was added. However, the .Card member now seems to be null in the source object. I also noticed that the list of sources on the customer object is of a different type: Source instead of StripeSource.

I wonder, is this inherent behaviour from Stripe or could it be something missing in Stripe.net?

```c#
StripeSourceService sourceService = new StripeSourceService("sk_myPrivateApiKey");
StripeCustomerService customerService = new StripeCustomerService("sk_myPrivateApiKey");
StripeSource source = sourceService.Get("src_theSourceId");
// source.Card is an object with card information
customerService.Update("cus_theCustomerId", new StripeCustomerUpdateOptions() {  DefaultSource = "src_theSourceId" });
StripeCustomer customer = customerService.Get("cus_theCustomerId");
Source customerSource = customer.Sources.Where(x => x.Id == "src_theSourceId").First();
// customerSource.Card is null
```

(If this is expected behaviour, I apologize for posting an issue here. I have been working with Stripe and Stripe.net for a while, but I am by no means a Stripe pro.)

Regards,
Bendik Engebretsen

**Comment by ob-stripe** [comment]
Hi @BendicoE, thanks for reporting this.

There are two different classes with very similar names but different behaviors:
- `StripeSource` represents a [source object](https://stripe.com/docs/api#source_object)
- `Source` is a wrapper class that is used to represent any customer's [payment source](https://stripe.com/docs/api#customer_object-sources), which may be a card (`StripeCard`), a bank account (`StripeBankAccount`) or... a source (`StripeSource`)

So you'd need to modify your code like this:
```c#
Source customerSource = customer.Sources.Where(x => x.Id == "src_theSourceId").First();
StripeSource customerCardSource = source.Source;
// customerCardSource.Card will have the values you expect
```

However, that won't work either because right now, `Source` does not support wrapping `StripeSource`. This is an issue with the library, we will push a fix and release a new version shortly.

**Comment by BendicoE** [comment]
Thanks, Oliver!
I think I understand the class relations better now:-) One little thing: I cannot not seem to find any Source member on the source object - is this because of the current issue with the library that you mention?

The reason I'm digging into this, is because I want to check the Fingerprint of a supplied source card against the Fingerprints of existing source cards the customer has used before. (To avoid attaching the same card multiple times to the customer.) This used to be straightforward previously when using the card workflow. Is it possible to do the same type of checking when using the the source workflow?

**Comment by ob-stripe** [comment]
Yes, currently the `Source` class won't let you access `StripeSource` objects.

As a workaround, you can list all `StripeSource` objects attached to a customer like this:
```c#
var SourceListOptions = new StripeSourceListOptions
{
    Type = StripeSourceType.Card
};
StripeList<StripeSource> cardSources = sourceService.List(Customer.Id, SourceListOptions);
```

This will return a list of `StripeSource` objects, and you can then access the `Card` property on each object.

**Comment by BendicoE** [comment]
Ah, great! You've just solved my problem, thank you so much!

**Comment by ob-stripe** [comment]
@BendicoE I just released v12.1.0 with a fix for your issue. When using source objects, you will now be able to access them via the `SourceObject` property in `Source` wrappers, e.g. like so:

```c#
// creating a charge with a source object of type card
StripeCharge = chargeService.Create(chargeCreateOptions);
var cardBrand = Charge.Source.SourceObject.Card.Brand;
```

v12.1.0 should be released on NuGet within the hour.

**Comment by BendicoE** [comment]
Super, thanks a lot @ob-stripe ! I'll be implementing this asap.
